otter/integration/lib/cloud_load_balancer.py

```diff
@@ -0,0 +1,161 @@
+        """Returns the JSON structure (as a Python dictionary) used to
+        configure the cloud load balancer via API operations.
+        """
+        return {
+            "loadBalancer": {
+                "name": "a-load-balancer",
+                "port": 80,
+                "protocol": "HTTP",
+                "virtualIps": [{
+                    "type": "PUBLIC",
+                    "ipVersion": "IPV6",
+                }],
+            }
+        }
+
+    def get_state(self, rcs):
+        """Returns the current state of the cloud load balancer.
+
+        :param TestResources rcs: The resources used to make appropriate API
+            calls with.
+        :return: A `Deferred` which, when fired, returns the parsed JSON for
+            the current cloud load balancer state.
+        """
+        return (
+            treq.get(
+                "%s/loadbalancers/%s" % (
+                    str(rcs.endpoints["loadbalancers"]),
+                    self.clb_id
+                ),
+                headers=headers(str(rcs.token)),
+                pool=self.pool,
+            )
+            .addCallback(check_success, [200])
+            .addCallback(treq.json_content)
+        )
+
+    def wait_for_state(
+        self, rcs, state_desired, timeout, period=10, clock=None
+    ):
+        """Waits for the cloud load balancer to reach a certain state.  After
+        a timeout, a `TimeoutError` exception will occur.
+
+        :param TestResources rcs: The resources used to make appropriate API
+            calls with.
+        :param str state_desired: The state you expect the cloud load balancer
+            to eventually reach.
+        :param int timeout: The number of seconds to wait before timing out.
+        :param int period: The number of seconds between polls to the cloud
+            load balancer.  If left unspecified, it defaults to 10 seconds.
+        :param twisted.internet.interfaces.IReactorTime clock: If provided,
+            the clock to use for scheduling things.  Defaults to `reactor`
+            if not specified.
+        :result: A `Deferred` which if fired, returns the same test resources
+            as provided this method.  This signifies the state has been
+            reached.  If the state has not been attained in the timeout period,
+            an exception will be raised, which can be caught in an Errback.
+        """
+        def check(state):
+            lb_state = state["loadBalancer"]["status"]
+            if lb_state == state_desired:
+                return rcs
+
+            raise TransientRetryError()
+
+        def poll():
+            return self.get_state(rcs).addCallback(check)
+
+        return retry_and_timeout(
+            poll, timeout,
+            can_retry=transient_errors_except(),
+            next_interval=repeating_interval(period),
+            clock=clock or reactor,
+            deferred_description=(
+                "State {} not reached in {} s timeout.".format(
```

Comment:
The deferred description is more of a description of what kind of work is being timed out - the [TimedOutException](https://github.com/rackerlabs/otter/blob/master/otter/util/deferredutils.py#L56) already formats it with the correct timeout time.  So possibly just something like `"Waiting for cloud load balancer to reach {0} state".format(state_desired)`?